sympy/core/mul.py

```diff
@@ -1222,7 +1222,10 @@ def _eval_is_prime(self):
             r = S.One
             for arg in self.args:
                 r *= arg
-            return r.is_prime
+            if self != r:
+                return r.is_prime
+            else:
+                return None
```

Comment:
Avoid negation in conditions if possible, so here this gives:

```
if self == r:
    return None
else:
    return r.is_prime
```

This is also a the "check early exit conditions" pattern; my personal recommendation is to code it as follows because it's just as easy to read, allows adding and reordering exit conditions without having to re-indent code, and avoids burying the standard case in deeply nested `if`s:

```
if self == r:
    return None
return r.is_prime
```